ShippingZones.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  ActivityIndicator,
+  Button,
+  FlatList,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from "react-native";
+import React, { useEffect, useState } from "react";
+import { fetchShippingZones, ShippingZone } from "./API/ShippingZoneAPI";
+import { Dependency, storeDependency } from "./Storage/AppDependencies";
+import { useNavigate } from "react-router";
+
+type RowProps = {
+  title: string;
+  body: string;
+  caption: string;
+};
+
+function Row(props: RowProps): JSX.Element {
+  return (
+    <View style={styles.row}>
+      <Text style={styles.row.title}> {props.title} </Text>
+      <Text style={styles.row.body}> {props.body} </Text>
+      <Text style={styles.row.caption}> {props.caption} </Text>
+    </View>
+  );
+}
+
+const ShippingZonesList = () => {
+  const [isLoading, setLoading] = useState(false);
+  const [data, setData] = useState<ShippingZone[]>([]);
+
+  const navigate = useNavigate();
+
+  /*
+   * Fetches the neccessary data for the shipping zones list.
+   */
+  const fetchData = async () => {
+    setLoading(true);
+
+    const zones = await fetchShippingZones();
+
+    setLoading(false);
+    setData(zones);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      {isLoading ? (
+        <SafeAreaView>
+          <ActivityIndicator />
+        </SafeAreaView>
+      ) : (
+        <FlatList
+          contentInsetAdjustmentBehavior="always"
+          style={styles.list}
+          data={data}
+          renderItem={({ item }) => (
+            <Row
+              title={item.title}
+              body={item.locations.map((location) => location.code).join(" - ")}
+              caption={item.methods.map((method) => method.title).join(" - ")}
+            />
+          )}
+          keyExtractor={(item) => item.id}
+        />
+      )}
+      <SafeAreaView>
+        <Button
+          title="+ Add new Shipping Zone"
+          onPress={() => navigate("/addShippingZone")}
```

Comment:
Do you think is worth having these paths in some centralized place? It seems easy to mistype them 😅